File: Elixir/p002_basic_neuron_layer.ex
defmodule P002BasicNeuronLayer do
  @moduledoc """
  Create a basic neuron layer in Elixir

  Associated tutorial https://www.youtube.com/watch?v=lGLto9Xd7bU
  """

  import Enum, only: [at: 2]

  def calculate() do
    inputs = [1.0, 2.0, 3.0, 2.5]
    weights1 = [0.2, 0.8, -0.5, 1.0]
    weights2 = [0.5, -0.91, 0.26, -0.5]
    weights3 = [-0.26, -0.27, 0.17, 0.87]
    bias1 = 2.0
    bias2 = 3.0
    bias3 = 0.5

    [at(inputs, 0) * at(weights1, 0) + at(inputs, 1) * at(weights1, 1) + at(inputs, 2) * at(weights1, 2) + at(inputs, 3) * at(weights1, 3) + bias1,
     at(inputs, 0) * at(weights2, 0) + at(inputs, 1) * at(weights2, 1) + at(inputs, 2) * at(weights2, 2) + at(inputs, 3) * at(weights2, 3) + bias2,
     at(inputs, 0) * at(weights3, 0) + at(inputs, 1) * at(weights3, 1) + at(inputs, 2) * at(weights3, 2) + at(inputs, 3) * at(weights3, 3) + bias3]
    |> IO.inspect
  end

end

File: VB.NET/p002-Basic-Neuron-Layer.vb
Module p002_Basic_Neuron_Layer
    Sub Main(args As String())
        Dim inputs() As Double = {1.0, 2.0, 3.0, 2.5}
        Dim weights1() As Double = {0.2, 0.8, -0.5, 1.0}
        Dim weights2() As Double = {0.5, -0.91, 0.26, -0.5}
        Dim weights3() As Double = {-0.26, -0.27, 0.17, 0.87}

        Dim bias1 As Double = 2.0
        Dim bias2 As Double = 3.0
        Dim bias3 As Double = 0.5

        Dim output() As Double = {(inputs(0) * weights1(0) + inputs(1) * weights1(1) + inputs(2) * weights1(2) + inputs(3) * weights1(3) + bias1),
                                  (inputs(0) * weights2(0) + inputs(1) * weights2(1) + inputs(2) * weights2(2) + inputs(3) * weights2(3) + bias2),
                                  (inputs(0) * weights3(0) + inputs(1) * weights3(1) + inputs(2) * weights3(2) + inputs(3) * weights3(3) + bias3)}

        Console.WriteLine($"[{String.Join(", ", output)}]")
    End Sub
End Module
